PR: https://github.com/odoo/odoo/pull/

From: be169f6a8181bcd2731ba804f1b21fa8da9eb210
From: Simon Lejeune
Date: 2020-01-31 10:22:14

Structural Changes: 4
Total Changes: 35

[REF] stock_picking_batch: stock picking to its own file

task-2069646

================================= pseudo patch: =================================

--- a/addons/stock_picking_batch/models/__init__.py
+++ b/addons/stock_picking_batch/models/__init__.py
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from . import stock_picking
 from . import stock_picking_batch

--- a/None
+++ b/addons/stock_picking_batch/models/stock_picking.py
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    batch_id = fields.Many2one(
+        'stock.picking.batch', string='Batch Transfer',
+        check_company=True,
+        states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
+        help='Batch associated to this transfer', copy=False)
+
+    def _should_show_transfers(self):
+        if len(self.batch_id) == 1 and self == self.batch_id.picking_ids:
+            return False
+        return super()._should_show_transfers()

--- a/addons/stock_picking_batch/models/stock_picking_batch.py
+++ b/addons/stock_picking_batch/models/stock_picking_batch.py
@@ -75,18 +75,3 @@ class StockPickingBatch(models.Model):
             return self.env.ref('stock_picking_batch.mt_batch_state')
         return super(StockPickingBatch, self)._track_subtype(init_values)
 
-
-class StockPicking(models.Model):
-    _inherit = "stock.picking"
-
-    batch_id = fields.Many2one(
-        'stock.picking.batch', string='Batch Transfer',
-        check_company=True,
-        states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
-        help='Batch associated to this transfer', copy=False)
-
-    def _should_show_transfers(self):
-        if len(self.batch_id) == 1 and self == self.batch_id.picking_ids:
-            return False
-        return super()._should_show_transfers()
-
